src/Site/SiteAwareTrait.php

```diff
@@ -28,96 +31,86 @@ public function setSites(Sites $sites)
     {
-        if (empty($site_env_id)) {
-            return [null, null];
+        if (false === strpos($site_env, '.')) {
+            throw new TerminusException('The environment argument must be given as <site_name>.<environment>');
         }
 
-        list($site_id, $env_id) = array_pad(explode('.', $site_env_id), 2, null);
+        $site_env_parts = explode('.', $site_env);
+        $site_id = $site_env_parts[0];
+        $env_id = $site_env_parts[1];
 
-        $site = $this->getSite($site_id);
-        $env = !empty($env_id) ? $site->getEnvironments()->get($env_id) : null;
-        return [$site, $env];
+        return $this->sites()->get($site_id)->getEnvironments()->get($env_id);
     }
 
     /**
-     * Get the site and environment with the given ids.
-     *
-     * @TODO This should be moved to the input/validation stage when that is available.
+     * @param string $site_env
+     * @return TerminusModel|null
      *
-     * @param string  $site_env_id The site/environment id in the form <site>[.<env>]
-     * @param string  $default_env The default environment to use if none is specified
-     * @return array  The site and environment in an array.
-     *
-     * @deprecated
-     * Use $this->getSite($site_id | $site_env)->getEnvironments()->get($env)
-     * for a typed object return.
-     *
-     * @throws TerminusException
+     * @throws \Pantheon\Terminus\Exceptions\TerminusException
+     * @throws \Pantheon\Terminus\Exceptions\TerminusNotFoundException
      */
-    public function getSiteEnv($site_env_id, $default_env = null)
+    public function getOptionalEnv(string $site_env): ?TerminusModel
     {
-        list($site_id, $env_id) = array_pad(explode('.', $site_env_id), 2, null);
-        $env_id = !empty($env_id) ? $env_id : $default_env;
-
-        if (empty($site_id) || empty($env_id)) {
-            throw new TerminusException('The environment argument must be given as <site_name>.<environment>');
+        if (false === strpos($site_env, '.')) {
+            return null;
         }
 
-        $site = $this->getSite($site_id);
-        $env = $site->getEnvironments()->get($env_id);
-        return [$site, $env];
+        return $this->getEnv($site_env);
     }
 
     /**
-     * Get the site and environment with the given IDs, provided the site is not frozen.
+     * Verifies the site is not in the frozen state, throws an exception otherwise.
      *
-     * @TODO This should be moved to the input/validation stage when that is available.
+     * @param string $site_env
      *
-     * @param string  $site_env_id The site/environment id in the form <site>[.<env>]
-     * @param string  $default_env The default environment to use if none is specified
-     * @return array  The site and environment in an array.
-     * @throws TerminusException
+     * @throws \Pantheon\Terminus\Exceptions\TerminusException
      */
-    public function getUnfrozenSiteEnv($site_env_id, $default_env = null)
```

Comment:
Terminus plugins might use `getUnfrozenSiteEnv` and other methods removed from this trait, and we do not want to break them in Terminus 3. Please restore all of the removed method signatures, mark them `@deprecated`, and have them call through to the new preferred methods.

Reply:
fixed